tests/unit/config/test_config.py

```diff
@@ -447,6 +447,30 @@ def test_master_pillar_roots_glob(self):
             if os.path.isdir(tempdir):
                 shutil.rmtree(tempdir)
 
+    def test_master_id_function(self):
+        try:
+            tempdir = tempfile.mkdtemp(dir=TMP)
+            master_config = os.path.join(tempdir, 'master')
```

Comment:
All the other test cases in this file do it the way I have done it. For the sake of consistency, I will leave it the way it is. If you believe that this approach is not the best approach in general, perhaps a separate PR that changes all of the test cases is a better approach. Changing the way the tests are executed in this file is beyond the scope of this PR.